
<!---
The page title should not go in the menu
-->
<p class="page-title">File Formats</p>

!!! Danger " "
    Clicking on the file format links in the sections below will open up a new tab and take you to the IGV Desktop documentation, where the file formats are described. Note, you may see some file formats in the IGV Desktop documentation that are not listed below; not all of the file formats supported by IGV Desktop are supported by IGV-Web.
    
## Data Tracks


* <a href="https://igv.org/doc/desktop/#FileFormats/DataTracks/#bam" target="_blank">BAM </a>
* <a href="https://igv.org/doc/desktop/#FileFormats/DataTracks/#bed" target="_blank">BED </a>
* <a href="https://igv.org/doc/desktop/#FileFormats/DataTracks/#bedgraph" target="_blank"> bedGraph </a>
* <a href="https://igv.org/doc/desktop/#FileFormats/DataTracks/#bedpe" target="_blank"> BEDPE </a>
* <a href="https://igv.org/doc/desktop/#FileFormats/DataTracks/#bigbed" target="_blank"> bigBed </a>
* <a href="https://igv.org/doc/desktop/#FileFormats/DataTracks/#bigwig" target="_blank"> bigWig </a>
* <a href="https://igv.org/doc/desktop/#FileFormats/DataTracks/#broadpeak" target="_blank"> broadPeak </a>
* <a href="https://igv.org/doc/desktop/#FileFormats/DataTracks/#bp-rna-base-pairing" target="_blank">BP (RNA secondary structure) </a>
* <a href="https://igv.org/doc/desktop/#FileFormats/DataTracks/#cram" target="_blank"> CRAM </a>
* <a href="https://igv.org/doc/desktop/#FileFormats/DataTracks/#genepred" target="_blank"> genePred </a>
* <a href="https://igv.org/doc/desktop/#FileFormats/DataTracks/#genepredext" target="_blank"> genePredExt </a>
* <a href="https://igv.org/doc/desktop/#FileFormats/DataTracks/#gffgtf" target="_blank"> GFF3 </a>
* <a href="https://igv.org/doc/desktop/#FileFormats/DataTracks/#gffgtf" target="_blank"> GTF </a>
* <a href="https://igv.org/doc/desktop/#FileFormats/DataTracks/#gwas" target="_blank"> GWAS </a>
* interact 
* <a href="https://igv.org/doc/desktop/#FileFormats/DataTracks/#maf-mutation-annotation-format" target="_blank">MAF (mutation annotation format) </a>
* <a href="https://igv.org/doc/desktop/#FileFormats/DataTracks/#mut" target="_blank"> MUT </a>
* <a href="https://igv.org/doc/desktop/#FileFormats/DataTracks/#narrowpeak" target="_blank"> narrowPeak </a>
* <a href="https://igv.org/doc/desktop/#FileFormats/DataTracks/#seg" target="_blank"> SET </a>
* <a href="https://igv.org/doc/desktop/#FileFormats/DataTracks/#tdf" target="_blank"> TDF </a>
* <a href="https://igv.org/doc/desktop/#FileFormats/DataTracks/#vcf" target="_blank"> VCF </a>
* <a href="https://igv.org/doc/desktop/#FileFormats/DataTracks/#wig" target="_blank"> WIG </a>

## Sample Info (Attributes)

Sample information includes three types of information:

* The **attributes** and their values for each sample - Required.
* **Sample mapping** information - Needed only if the sample names in the attribute information do not match the sample names in the data files.
* **Colors** to use in the attribute display - Needed only if the you wish to specify colors that are not the same as the defaults.

The file format is described <a href="https://igv.org/doc/desktop/#FileFormats/SampleInfo/" target="_blank"> here</a>.

## Genomes

* <a href="https://igv.org/doc/desktop/#FileFormats/Genomes/#fasta" target="_blank">FASTA </a>
* <a href="https://igv.org/doc/desktop/#FileFormats/Genomes/#igv-reference-genome-json" target="_blank">IGV reference genome (JSON) </a>
* <a href="https://igv.org/doc/desktop/#FileFormats/Genomes/#cytoband" target="_blank"> Cytoband </a>


